launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -352,6 +358,33 @@ void FirstLaunchView::extractGogData()
 		QFile(fileExe).copy(tmpFileExe);
 		QFile(fileBin).copy(tempDir.filePath("h3_gog-1.bin"));
 
+		QString errorText{};
+
+		auto isGogGalaxyExe = [](QString fileExe) {
+			std::ifstream is(fileExe.toStdString(), std::ios::binary);
+			if (!is)
+				return false;
+			is >> std::noskipws;
+			is.seekg(0, std::ios::end);
+			std::streampos fileSize = is.tellg();
+			is.seekg(0, std::ios::beg);
+
+			if(fileSize > 10 * 1024 * 1024)
+				return false; // avoid to load big files; galaxy exe is smaller...
+
+			std::vector<char> buffer;
+			buffer.reserve(fileSize);
+			buffer.insert(buffer.begin(), std::istream_iterator<char>(is), std::istream_iterator<char>());
+
+			std::array<char, 20> magic_id{ 0x47, 0x00, 0x4F, 0x00, 0x47, 0x00, 0x20, 0x00, 0x47, 0x00, 0x61, 0x00, 0x6C, 0x00, 0x61, 0x00, 0x78, 0x00, 0x79, 0x00 }; //GOG Galaxy
+
+			auto res = std::search(buffer.begin(), buffer.end(), magic_id.begin(), magic_id.end());
```

Comment:
is the header located at a random place and not at 0x0?

Reply:
Yes. At a "random" position. It isn't really a header. It's a String containing "GOG Galaxy". So basically this way it should work with newer versions too.

Reply:
> It's a String containing "GOG Galaxy"

then in code you should use something like `u"GOG Galaxy"` to get utf-16 string and then view it as a byte array